### DISCUSSION TITLE
How to set fileype for nui.popup (to get syntax highlighting)?

### DISCUSSION THREAD

**Discussion by darzok0914** [MunifTanjim/nui.nvim#77] [Q&A] [answered]
Hi, I think everything is in the title, it would be great if we could use the popup window to display some code for instance. it would be nice to get the color highlighting from tree sitter by being able to associate a file type to the popup window, I went though the documentation and didn't see anything like this ?

**Comment by MunifTanjim** [comment, marked as answer]
So, `filetype` is just a buffer option (check `:help 'filetype'`). You can do this to set it:

```lua
local popup = Popup({
  ...
  buf_options = {
    ...
    filetype = "lua",
  },
  ...
})
```

or

```lua
vim.api.nvim_buf_set_option(popup.bufnr, "filetype", "lua")
```

Either of them will do the trick.

**Comment by darzok0914** [comment]
thanks for your response, it did the trick indeed :)
